kore/src/SMT.hs

```diff
@@ -259,89 +249,70 @@ different threads may be interleaved; use 'inNewScope' to acquire exclusive
 access to the solver for a sequence of commands.
 
  -}
-newtype SmtT m a = SmtT { runSmtT :: ReaderT (MVar Solver) m a }
+newtype SMT a = Smt { runSmt :: ReaderT (MVar Solver) (Log.LoggerT IO) a }
     deriving
-        ( Applicative
-        , Functor
-        , Monad
-        , MonadCatch
-        , MonadIO
-        , MonadThrow
-        , Morph.MFunctor
+        ( Applicative, Functor, Monad, Reader.MonadReader (MVar Solver)
+        , MonadCatch, MonadIO, MonadThrow
+        , MonadLog
         )
 
-instance MonadUnliftIO m => MonadUnliftIO (SmtT m) where
+instance MonadUnliftIO SMT where
     askUnliftIO =
-        SmtT $ ReaderT $ \r ->
+        Smt $ ReaderT $ \r ->
             withUnliftIO $ \u ->
-                return (UnliftIO (unliftIO u . flip runReaderT r . runSmtT))
-
-type SMT = SmtT IO
+                return (UnliftIO (unliftIO u . flip runReaderT r . runSmt))
 
-withSolver' :: (Solver -> IO a) -> SMT a
-withSolver' action = SmtT $ do
+withSolver'
```

Comment:
If we follow my suggestions for the `SimpleSMT` module, then this can be specialized to:

```.hs
withSolver' :: (Solver -> IO a) -> SMT a
```

and we can remove the dependency on `UnliftIO`.